osbs/tekton.py

```diff
@@ -518,34 +518,22 @@ def get_error_message(self):
             if stats['status']['conditions'][0]['reason'] == 'Succeeded':
                 continue
 
-            err_message += f"pipeline task '{task_name}' failed:\n"
-
             if 'steps' in stats['status']:
                 for step in stats['status']['steps']:
                     if 'terminated' in step:
                         exit_code = step['terminated']['exitCode']
                         if exit_code == 0:
                             continue
 
-                        reason = step['terminated']['reason']
-                        err_message += f"task step '{step['name']}' failed with exit " \
-                                       f"code: {exit_code} " \
-                                       f"and reason: '{reason}'\n"
-                    else:
-                        err_message += f"task step '{step['name']}' is missing 'terminated' key " \
-                                       f"with exit code and reason\n"
-
-            else:
-                task_condition = stats['status']['conditions'][0]
-                err_message += f"task run '{task_name}' failed with reason:" \
-                               f" '{task_condition['reason']}' and message:" \
-                               f" '{task_condition['message']}'\n"
-
-        if not task_runs_status:
-            pipeline_run_condition = data['status']['conditions'][0]
-            err_message += f"pipeline run {self.pipeline_run_name} failed with reason:" \
-                           f" '{pipeline_run_condition['reason']}' and message:" \
-                           f" '{pipeline_run_condition['message']}'\n"
+                        if 'message' in step['terminated']:
+                            try:
+                                message_json = json.loads(step['terminated']['message'])
+                                for message in message_json:
+                                    if message['key'] == 'task_result':
+                                        err_message += f"Error in {task_name}: " \
+                                                       f"{message['value']};\n"
+                            except Exception:
+                                continue
```

Comment:
Does this have to catch the generic `Exception`?

Should we at least log the exception?

Reply:
I don't want it to fail here for any reason, because then it would fail for koji-c as well and we wouldn't get any useful messages

can log it